src/backends/policy_engine/souffle/utils.cc

```diff
@@ -0,0 +1,47 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/backends/policy_engine/souffle/utils.h"
+
+#include <fstream>
+
+#include "absl/strings/str_format.h"
+
+namespace raksha::backends::policy_engine::souffle {
+
+absl::Status WriteFactsStringToFactsFile(
```

Comment:
If you add an API, we should add tests too. This could probably stay as a static method of RakshaDatalogFacts?

Reply:
The policy rules aren't part of `RakshaDatalogFacts`, so in order to share code like this I have to make this file public. So I guess I'll write some tests for it.

Reply:
Wrote tests for this function.